docs/source/preprocessing.mdx

```diff
@@ -204,15 +212,312 @@ To double-check what is fed to the model, we can decode each list in _input_ids_
+          [-0.0118, -0.0039, -0.0039,  ..., -0.0980, -0.0980, -0.1529]],
+ 
+         [[ 0.3961,  0.4431,  0.4980,  ..., -0.9216, -0.9137, -0.9216],
+          [ 0.3569,  0.4510,  0.5216,  ..., -0.9059, -0.9137, -0.9137],
+          [ 0.4118,  0.4745,  0.5216,  ..., -0.9137, -0.8902, -0.7804],
+          ...,
+          [-0.2314, -0.1922, -0.2078,  ..., -0.4196, -0.4275, -0.3882],
+          [-0.1843, -0.1686, -0.2000,  ..., -0.3647, -0.3804, -0.4039],
+          [-0.1922, -0.1922, -0.1922,  ..., -0.2941, -0.2863, -0.3412]]])}
+```
+
+## Multimodal
+
+Finally, for multimodal tasks you will use a combination of everything you've learned so far. In this multimodal preprocessing tutorial, you will focus on a automatic speech recognition (ASR) task. This means you will need a:
+
+* Feature extractor to preprocess the audio data.
+* Tokenizer to process the text.
+
+Let's use the [`TIMIT`](https://huggingface.co/datasets/timit_asr) dataset:
+
+```py
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("timit_asr", split="test")
+```
+
+For this tutorial, you are mainly interested in the `audio` and `text` column. Remove all the other columns:
+
+```py
+>>> dataset = dataset.map(remove_columns=["file", "phonetic_detail", "word_detail", "dialect_region", "sentence_type", "speaker_id", "id"])
+```
+
+Now let's take a look at the `audio` and `text` columns:
+
+```py
+>>> dataset[0]["audio"]
+{'array': array([ 2.4414062e-04, -3.0517578e-05,  3.0517578e-05, ...,
+         6.1035156e-05,  9.1552734e-05,  6.1035156e-05], dtype=float32),
+ 'path': '/root/.cache/huggingface/datasets/downloads/extracted/404950a46da14eac65eb4e2a8317b1372fb3971d980d91d5d5b221275b1fd7e0/data/TEST/DR4/MGMM0/SX139.WAV',
+ 'sampling_rate': 16000}
+
+>>> dataset[0]["text"]
+'The bungalow was pleasantly situated near the shore.'
+```
+
+### Processor
+
+A processor combines the feature extractor and tokenizer. Load a processor with the [`AutoProcessor.from_pretrained] method:
+
+```py
+>>> from transformers import AutoProcessor
+
+>>> processor = AutoProcessor.from_pretrained("facebook/wav2vec2-base-960h")
+```
+
+1. Create a function to process the audio data to `input_values`, and to tokenize the text to `labels`. These are your inputs to the model:
+
+```py
+>>> def prepare_dataset(example):
+...    audio = example["audio"]
+...
+...    example["input_values"] = processor(audio["array"], sampling_rate=16000)
+...    
+...    with processor.as_target_processor():
+...            example["labels"] = processor(example["text"]).input_ids
+...    return example
+```
+
+2. Apply `prepare_dataset` to your example, and you will see the processor has added `input_values` and `labels`:
+
+```py
+>>> prepare_dataset(dataset[0])
+```
+
+Awesome, you should now be able to preprocess data for any modality! In the next tutorial, you will learn how to train a model on the preprocessed data.
+
 ## Everything you always wanted to know about padding and truncation
```

Comment:
This could also go in a more advanced tutorial.

Reply:
I was thinking something similar, and think it'd be a good fit for the Concept Guide section when we have it. 😄